pkg/config/store.go

```diff
@@ -0,0 +1,324 @@
+	// e.g., "fetch-7b8c9d5f4-xyz12" -> "fetch"
+	parts := strings.Split(hostname, "-")
+	if len(parts) > 0 {
+		serverName := parts[0]
+		logger.Debugf("Extracted server name '%s' from hostname '%s'", serverName, hostname)
+		return serverName
+	}
+
+	logger.Debugf("Could not extract server name from hostname '%s'", hostname)
+	return ""
+}
+
+// NewKubernetesStore creates a new Kubernetes ConfigMap-based configuration store
+// If serverName is empty, it will attempt to detect it from the hostname
+func NewKubernetesStore(serverName string) (*KubernetesStore, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get in-cluster config: %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %w", err)
+	}
+
+	// Auto-detect server name from hostname if not provided
+	if serverName == "" {
+		serverName = getMCPServerNameFromHostname()
+	}
+
+	namespace := getCurrentNamespace()
+
+	store := &KubernetesStore{
+		client:     clientset,
+		namespace:  namespace,
+		serverName: serverName,
+	}
+
+	logger.Infof("Using Kubernetes namespace '%s' for configuration storage (server: %s)", namespace, serverName)
+
+	return store, nil
+}
+
+// getConfigMapName returns the ConfigMap name for this store
+// If serverName is provided, it creates a server-specific name
+func (s *KubernetesStore) getConfigMapName() string {
+	if s.serverName != "" {
+		return fmt.Sprintf("%s-%s", ConfigMapName, s.serverName)
+	}
+	return ConfigMapName
+}
+
+// Load loads configuration from Kubernetes ConfigMap
+func (s *KubernetesStore) Load(ctx context.Context) (*Config, error) {
+	configMapName := s.getConfigMapName()
+	configMap, err := s.client.CoreV1().ConfigMaps(s.namespace).Get(ctx, configMapName, metav1.GetOptions{})
+	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			// Create default config if ConfigMap doesn't exist
+			logger.Infof("Config ConfigMap not found, creating default configuration")
+			config := createNewConfigWithDefaults()
+			err = s.Save(ctx, &config)
+			if err != nil {
+				return nil, fmt.Errorf("failed to save default config: %w", err)
+			}
+			return &config, nil
+		}
+		return nil, fmt.Errorf("failed to get config ConfigMap: %w", err)
+	}
+
+	configYAML, exists := configMap.Data[ConfigMapKey]
+	if !exists {
+		return nil, fmt.Errorf("config key %s not found in ConfigMap", ConfigMapKey)
+	}
+
+	var config Config
+	err = yaml.Unmarshal([]byte(configYAML), &config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config YAML: %w", err)
+	}
+
+	// Apply backward compatibility fixes
+	err = applyBackwardCompatibility(&config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to apply backward compatibility fixes: %w", err)
+	}
+
+	return &config, nil
+}
+
+// Save saves configuration to Kubernetes ConfigMap
+func (s *KubernetesStore) Save(ctx context.Context, config *Config) error {
```

Comment:
How come we are running save here? I thought the Operator is the one that does the creation of the ConfigMap?